dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedIntervalIterableTest.java

```diff
@@ -172,12 +184,12 @@ public void testForLogStartTimeAndLogEndTime() throws Throwable {
         DateTimeFormatter.ofPattern(ConnectorArguments.ZonedParser.DEFAULT_PATTERN, Locale.US);
 
     LocalDateTime requestedStartParsed = LocalDateTime.parse(requestedStart, formatter);
-    LocalDateTime requestedEndParsed = LocalDateTime.parse(requestedEnd, formatter);
+    LocalDateTime expectedEndParsed = LocalDateTime.parse("2020-07-05 10:17:00", formatter);
 
     ZonedDateTime expectedStart =
-        requestedStartParsed.atZone(ZoneOffset.UTC).truncatedTo(ChronoUnit.HOURS);
+        requestedStartParsed.atZone(ZoneOffset.UTC).truncatedTo(ChronoUnit.SECONDS);
```

Comment:
I would recommend to avoid logic in tests. The expected values should be provided as literals, so that the test data (both input and output) are easy to understand without requiring any computations. Similarly in other tests.